### PR TITLE
Automate updating operator version for release

### PR DESCRIPTION
**Pull request by radTuti**
## Description

This handles [updating versions](https://github.com/tigera/operator/blob/master/RELEASING.md#updating-versions) when preparing for an operator release resulting in a PR similar to [this](https://github.com/tigera/operator/pull/2845/files#diff-224df6cb23a4afa3cbf26bfce45376e99e0f8db6b76e4b88191788ceea3d09e3)

## For PR author

- [x] Tests for change: Tested with private fork: https://github.com/tigera/operator/pull/2963
 
 ```sh
  git o release-v1.31 # checkout release branch
  make release-prep GIT_PR_BRANCH_BASE=master GIT_REPO_SLUG=tigera/operator CONFIRM=true   VERSION=v1.32.0 CALICO_VERSION=v3.26.1 CALICO_ENTERPRISE_VERSION=v3.18.0 COMMON_VERSION=v1.1.15
  ```

- [ ] ~If changing pkg/apis/, run `make gen-files`~
- [ ] ~If changing versions, run `make gen-versions`~

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [x] Milestone set according to targeted release.
- [x] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: Makefile
Comment (danudey) [inline Makefile:572]: We should ensure that `yq` is available and is the version that we want; e.g. we use yq2 (yq v2.x) in some projects, and the syntax between them is a bit different. I'd recommend checking that `yq` is available and that the version matches, and, if so, storing that in `$(YQ)` for use elsewhere.

(This has bitten me before in other scripts and Makefiles)

File: Makefile
Comment (danudey) [inline Makefile:943]: Another recommendation: define `CURL_CMD` near the top of the file as e.g. `curl -fsSL` (the same parameters you have here, plus 'be silent' and 'don't be silent if there's an error', so as to reduce output). That way we can easily override CURL_CMD for debugging (e.g. `make release-prep/create-pr CURL_CMD='curl -fvL'` or something).

File: Makefile
Comment (danudey) [inline Makefile:942]: The way you've implemented this is really elegant and clean, I'm a huge fan.

File: Makefile
Comment (danudey) [inline Makefile:596]: It would be nice if we could also automatically set the labels on the PR to e.g. 'docs-not-required' and 'release-note-not-required', since these docs and release note required are always defaulted to true and in this case they never are. This will reduce the work required to filter the release notes after the release is done.

File: Makefile
Comment (radTuti) [reply Makefile]: updated

File: Makefile
Comment (radTuti) [reply Makefile]: done

File: Makefile
Comment (radTuti) [reply Makefile:605]: i'm not sure the slash commands that marvin allows so i'll ask around to see if this will be a quick update; otherwise, it'll likely be another ticket to get that going

File: Makefile
Comment (rene-dekker) [inline Makefile:9]: Would it be possible to do everything with the yq & curl from go-build? I believe that's the yq we use in our other repos.

File: Makefile
Comment (rene-dekker) [inline Makefile:579]: I think we should refrain from the long abandoned term cnx.

File: Makefile
Comment (radTuti) [reply Makefile:9]: go-build does not have yq and curl. As operator does not currently import the Makefile from go-build I had a chat with Casey and the concensus was it was better not to import it ATM

Comment (caseydavenport) [reply Makefile:9]: We do still use go-build in this repo, we just don't use the common Makefile, and I don't think we should be adding a dependency on it.

File: Makefile
Comment (radTuti) [reply Makefile:602]: created [DE-2157](https://tigera.atlassian.net/browse/DE-2157) to address this

File: Makefile
Comment (rene-dekker) [reply Makefile:9]: > go-build does not have yq and curl
It has both
```
$ docker run -it -e LOCAL_USER_ID=1000  calico/go-build:v0.88 sh

$ curl --version
curl 7.74.0 ...

$ yq --version
yq version 3.4.1
```

**Comment by rene-dekker** [comment]
I think RELEASING.md should be updated to incorporate how an automated release can be performed and should include what the preconditions and results are.

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, squash the commits when I merge it, and delete the branch after I've merged it.

**Comment by marvin-tigera** [comment]
Removing "merge-when-ready" label due to new commits

File: RELEASING.md
Comment (rene-dekker) [inline RELEASING.md:41]: ensures*

**Review by rene-dekker — APPROVED** [review]
lgtm

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, squash the commits when I merge it, and delete the branch after I've merged it.

**Comment by marvin-tigera** [comment]
I'm sorry but I failed delete the branch after merging the pull request.

**Comment by tmjd** [comment]
This PR doesn't seem to have any provisions related to ensuring the Elasticsearch versions are updated, those versions do not match the Enteprise version. So I don't think this is ready to be used yet.

**Comment by tmjd** [comment]
@radTuti pointed out to me that Prometheus and Elasticsearch versions require a specific decision to update so will be updated when there is the work is scheduled to update those. I think also in my original review I missed that they were being skipped as part of the enterprise_versions.yaml.

**Pull request merged**
